api/src/main/java/org/openmrs/module/fhir2/providers/util/FhirProviderUtils.java

```diff
@@ -33,6 +36,21 @@ public static MethodOutcome buildCreate(DomainResource resource) {
 		return buildWithResource(methodOutcome, resource);
 	}
 	
+	public static OperationOutcome buildDelete(OperationOutcome.IssueSeverity code, OperationOutcome.IssueType issueType,
+	        BaseServerResponseException exception) {
+		//created codeable concepts to be called by operationOutcome, codeableconcept only limited to two properties text and coding
+		CodeableConcept codeableConcept = new CodeableConcept().setText("MSG_DELETED").setText("informational")
+		        .setText("This resource has been deleted").setText("informational")
+		        .setText("http://terminology.hl7.org/CodeSystem/operation-outcome").setText("This resource has been deleted")
+		        .setText(exception.getMessage());
+		
+		OperationOutcome operationOutcome = new OperationOutcome();
+		operationOutcome.addIssue().setSeverity(OperationOutcome.IssueSeverity.INFORMATION).setCode(issueType)
+		        .setDetails(codeableConcept);// am assumming setting details will trigger only called codeableconcepts above             	
+		exception.setOperationOutcome(operationOutcome);
+		return operationOutcome;
+	}
```

Comment:
```suggestion
	public static OperationOutcome buildDelete(DomainResource resource) {
		OperationOutcome outcome = new OperationOutcome();
		outcome.addIssue().setSeverity(OperationOutcome.IssueSeverity.INFORMATION)
				.setCode(OperationOutcome.IssueType.INFORMATIONAL).setDetails(MSG_DELETED);
		return outcome;
	}
```

With `MSG_DELETED` defined like this:

```java
private static final CodeableConcept MSG_DELETED = new CodeableConcept();

static {
	MSG_DELETED.addCoding().setSystem("http://terminology.hl7.org/CodeSystem/operation-outcome")
			.setCode("MSG_DELETED").setDisplay("This resource has been deleted");
	MSG_DELETED.setText("This resource has been deleted");
}
```

Reply:
I  didnot add private access modifier since it triggered me to make MSG_DELETED  defined as a constant